Upstream: https://source.puri.sm/Librem5/calls/-/merge_requests/230

From 1b1588eb00fb716250c4875e82bc6d5f4e11bf58 Mon Sep 17 00:00:00 2001
From: Arnaud Ferraris <arnaud.ferraris@collabora.com>
Date: Tue, 12 Jan 2021 18:40:12 +0100
Subject: [PATCH 1/2] call-display: update libcallaudio API calls

Upstream modified the async API, make sure we use it.
---
 src/calls-call-display.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/calls-call-display.c b/src/calls-call-display.c
index 7d17772..2bfb8fc 100644
--- a/src/calls-call-display.c
+++ b/src/calls-call-display.c
@@ -211,7 +211,7 @@ stop_timeout (CallsCallDisplay *self)
 
 
 static void
-select_mode_complete (gboolean success, GError *error)
+select_mode_complete (gboolean success, GError *error, gpointer data)
 {
   if (error)
     {
@@ -261,11 +261,15 @@ call_state_changed_cb (CallsCallDisplay *self,
          state != CALLS_CALL_STATE_DIALING
          && state != CALLS_CALL_STATE_ALERTING);
 
-      call_audio_select_mode_async (CALL_AUDIO_MODE_CALL, select_mode_complete);
+      call_audio_select_mode_async (CALL_AUDIO_MODE_CALL,
+                                    select_mode_complete,
+                                    NULL);
       break;
 
     case CALLS_CALL_STATE_DISCONNECTED:
-      call_audio_select_mode_async (CALL_AUDIO_MODE_DEFAULT, select_mode_complete);
+      call_audio_select_mode_async (CALL_AUDIO_MODE_DEFAULT,
+                                    select_mode_complete,
+                                    NULL);
       break;
     }
 
-- 
2.26.2

